server/modules/user/controllers/get-user-dashboard.js

```diff
@@ -1,13 +1,38 @@
-import * as User from '../use-cases';
+import * as TherapistClient from '../../therapist-client/use-cases';
+import * as Programmes from '../../programme/use-cases';
 
 const getUserDashboard = async (req, res, next) => {
   const { user } = req;
+  const role = user.roles[0];
   try {
-    const results = await User.getUserDashboard({
-      id: user.id,
-      role: user.role,
-    });
-    res.json(results);
+    if (role === 'CLIENT') {
+      const therapist = await TherapistClient.getTherapistByClient({
+        id: user.id,
+      });
+
+      const programmes = await Programmes.getProgrammesByClient({
+        userId: user.id,
+      });
+
+      return res.json({ therapist, programmes });
+    }
+
+    if (role === 'THERAPIST') {
+      const clients = await TherapistClient.getClientsByTherapist({
+        id: user.id,
+      });
+
+      const protectedClients = clients.map((c) => ({
+        firstInitial: c.firstName[0],
+        lastInitial: c.lastName[0],
+        postcode: c.postcode,
```

Comment:
```suggestion
        postcode: c.postcode && c.postcode.slice(0, 2),
```

so in future if clients were able to add the full postcode it still protected

Reply:
Good idea. Done